### ISSUE TITLE
Action Workflow and global params do not work together

### ISSUE THREAD

**Issue by rklec** [microsoft/RulesEngine#626]
## STR

Let's say you have this action workflow:
```jsonc
{
  "$schema": "https://raw.githubusercontent.com/microsoft/RulesEngine/main/schema/workflow-schema.json",
  "WorkflowName": "MailEvaluationWorkflow",
  "GlobalParams": [
    {
      "Name": "myFavouriteThing",
      "Expression": "1"
    }
  ],
  // some action workflow rules here
```

### What happens

This will _always_ fail, if you have at least two rules, with an exception like this:
> Exception: Error while executing rule : SecondRuleName - An item with the same key has already been added. Key: myFavouriteThing, ActionResult.Output: null

### What should happen

IMHO this makes `GlobalParams` absolutely useless, as you cannot use them. It should just inject them once 

In a basic workflow this works fine, so this only affects the action workflow.

## More information

I have found https://github.com/microsoft/RulesEngine/issues/259, which seems to be about the same issue(?), where filtering etc. was introduced.
However, IMHO, this is still not a good way to handle it, is it?

Because come on, what use case does `GlobalParams` even have then? You cannot use them in any way if you have two rules, unless you do some filtering, which really is kinda silly to then always filter, is it?

## Workaround
Apparently just define all global params as "local params" in your first rule, which is passed to all others unless you filter.
This kinda makes them "global" IMHO and is not a good naming and for me at least was totally confusing. And I had to read and understand the issue here first to get why this happens.

## System

`RulesEngine` v5.0.3

**Comment by asulwer** [comment]
i would suggest posting this to a fork that is being [maintained](https://github.com/microsoft/RulesEngine/issues/604#issuecomment-2181168220)

**Comment by rklec** [comment]
Yeah, good point, did so: https://github.com/timophe-91/RulesEngine/issues/57 https://github.com/asulwer/RulesEngine/issues/116
